src/factories/authentication/custom_jwt_v1/custom-authentication-client.ts

```diff
@@ -48,11 +53,38 @@ export class CustomAuthenticationClient implements AuthenticationClient {
     };
   }
 
-  async isAuthenticated(accessToken: string): Promise<boolean> {
+  async isAuthenticated(accessToken: string): Promise<TokenPayload<User['id']>> {
     const { appConfig, tokenService } = this.dependencies;
     return tokenService.verifyToken(accessToken, appConfig.accessTokenConfig);
   }
 
+  async requestPasswordReset(username: string): Promise<boolean> {
+    const { usersRepository, securityService } = this.dependencies;
+    const [user, errOnSearch] = await handleAsync(usersRepository.findByUsername(username));
+
+    if (errOnSearch) {
+      throw new AppError('An error occurred while searching for existing user.', errOnSearch);
```

Comment:
I see that you are using same error handling code in multiple places, can we make one function in `util` to handle these cases?